app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -330,6 +347,28 @@ class HomeFragment : Fragment() {
             )
         }
 
+        with(requireContext().settings().showUnifiedSearchFeature) {
+            binding.searchSelector.isVisible = this
+            binding.searchEngineIcon.isGone = this
+        }
+
+        binding.searchSelector.apply {
+            setOnClickListener {
+                val orientation = if (context.settings().shouldUseBottomToolbar) {
+                    Orientation.UP
+                } else {
+                    Orientation.DOWN
+                }
+
+                UnifiedSearch.searchMenuTapped.record(mozilla.telemetry.glean.private.NoExtras())
+                searchSelectorMenu.menuController.show(anchor = it, orientation = orientation, forceOrientation = true)
+            }
+
+            setBackgroundResource(
+                context.theme.resolveAttribute(android.R.attr.selectableItemBackgroundBorderless),
+            )
+        }
```

Comment:
I have an interest to see more code refactored out of the `HomeFragment` and into their corresponding classes so that we could encapsulate all the logic around the search selector into the appropriate classes. Would this be possible here? Would it make sense add a public update function of some sort in `SearchSelectorMenu` or something else that can be called here?

Reply:
Actually, this might fit more with the `Builder` classes we did for the `HomeFragment`. See `TabCounterBuilder` and `HomeMenuBuilder`.

Reply:
We don't necessarily have to do this in this PR. This is more food for thought if we were able to encapsulate the logic here better so that our enormous `HomeFragment` won't hold so much of the logic related to various functions.

Reply:
Yes, I also think that it has to be encapsulated. Actually, it is – inside `SearchSelectorToolbarAction`, which is a `Toolbar.Action` and can't be used on a "fake" toolbar we have on the homescreen. Creating a more real toolbar for the `HomeFragment` could be a direction to go? I have little context why it was implemented that way.

Reply:
When I wrote `SearchSelectorToolbarAction`, it was for encapsulating the `Toolbar.Action` functionality that would be added to the edit toolbar actions. This is indeed a direction I am trying to get at, but I am getting the sense now that we should probably be a bit less concern about the ideas I presented in this conversation for the sake of this PR and we can revisit this more in a future cleanup of `HomeFragment`. We'll want to think about what makes sense to group up and where (should it be be done based on the component in the `binding`, etc). 

We can resolve this conversation.